prow/tide/tide.go

```diff
@@ -461,38 +461,37 @@ func (c *Controller) query() (map[string]PullRequest, error) {
 		} else {
 			queries = map[string]string{"": query.Query()}
 		}
-
 		for org, q := range queries {
-			org, q, i := org, q, i
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-				results, err := search(c.ghc.QueryWithGitHubAppsSupport, c.logger, q, time.Time{}, time.Now(), org)
-
-				resultString := "success"
-				if err != nil {
-					resultString = "error"
-				}
-				tideMetrics.queryResults.WithLabelValues(strconv.Itoa(i), org, resultString).Inc()
-
-				lock.Lock()
-				defer lock.Unlock()
-				if err != nil && len(results) == 0 {
-					c.logger.WithField("query", q).WithError(err).Warn("Failed to execute query.")
-					errs = append(errs, fmt.Errorf("query %d, err: %w", i, err))
-					return
-				}
-				if err != nil {
-					c.logger.WithError(err).WithField("query", q).Warning("found partial results")
-				}
-
-				for _, pr := range results {
-					prs[prKey(&pr)] = pr
-				}
-			}()
+			allQueries = append(allQueries, queryData{query: q, org: org, metricIndex: i})
 		}
 	}
-	wg.Wait()
+	queriesInParallel(
+		c.config().Tide.MaxGraphQLGoroutines,
+		allQueries,
+		func(q queryData) {
+			results, err := search(c.ghc.QueryWithGitHubAppsSupport, c.logger, q.query, time.Time{}, time.Now(), q.org)
+
+			resultString := "success"
+			if err != nil {
+				resultString = "error"
+			}
+			tideMetrics.queryResults.WithLabelValues(strconv.Itoa(q.metricIndex), q.org, resultString).Inc()
+
+			lock.Lock()
+			defer lock.Unlock()
+			if err != nil && len(results) == 0 {
+				c.logger.WithField("query", q).WithError(err).Warn("Failed to execute query.")
+				errs = append(errs, fmt.Errorf("query %d, err: %w", q.metricIndex, err))
+				return
+			}
+			if err != nil {
+				c.logger.WithError(err).WithField("query", q).Warning("found partial results")
+			}
+
+			for _, pr := range results {
+				prs[prKey(&pr)] = pr
```

Comment:
I know this predates the current PR, but why is this not racy?

Reply:
Probably the lock in 480 protects it? Maybe it's another Go thing, but I do not see the danger...